devicetypes/qubino/qubino-flush-dimmer-0-10V.src/qubino-flush-dimmer-0-10V.groovy

```diff
@@ -0,0 +1,671 @@
+			input (
+					title: it.name,
+					description: it.description,
+					type: "paragraph",
+					element: "paragraph"
+			)
+
+			switch(it.type) {
+				case "boolRange":
+					input(
+							name: it.key + "Boolean",
+							type: "bool",
+							title: "Enable",
+							description: "If you disable this option, it will overwrite setting below.",
+							defaultValue: it.defaultValue != it.disableValue,
+							required: false
+					)
+					input(
+							name: it.key,
+							type: "number",
+							title: "Set value (range ${it.range})",
+							defaultValue: it.defaultValue,
+							range: it.range,
+							required: false
+					)
+					break
+				case "boolean":
+					input(
+							type: "paragraph",
+							element: "paragraph",
+							description: "Option enabled: ${it.activeDescription}\n" +
+									"Option disabled: ${it.inactiveDescription}"
+					)
+					input(
+							name: it.key,
+							type: "bool",
+							title: "Enable",
+							defaultValue: it.defaultValue == it.activeOption,
+							required: false
+					)
+					break
+				case "enum":
+					input(
+							name: it.key,
+							title: "Select",
+							type: "enum",
+							options: it.values,
+							defaultValue: it.defaultValue,
+							required: false
+					)
+					break
+				case "range":
+					input(
+							name: it.key,
+							type: "number",
+							title: "Set value (range ${it.range})",
+							defaultValue: it.defaultValue,
+							range: it.range,
+							required: false
+					)
+					break
+			}
+		}
+		// Preferences template end
+	}
+}
+
+//Globals, input types used in sevice settings (parameter #1: Input 1 switch type)
+private getINPUT_TYPE_MONO_STABLE_SWITCH() {0}
+private getINPUT_TYPE_BI_STABLE_SWITCH() {1}
+private getINPUT_TYPE_POTENTIOMETER() {2}
+private getINPUT_TYPE_TEMPERATURE_SENSOR() {3}
+private getINPUT_TYPE_ILLUMINATION_SENSOR() {4}
+private getINPUT_TYPE_GENERAL_PURPOSE_SENSOR() {5}
+
+def installed() {
+	// Device-Watch simply pings if no device events received for 32min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+
+	// Preferences template begin
+	state.currentPreferencesState = [:]
+	parameterMap.each {
+		state.currentPreferencesState."$it.key" = [:]
+		state.currentPreferencesState."$it.key".value = getPreferenceValue(it)
+		if (it.type == "boolRange" && getPreferenceValue(it) == it.disableValue) {
+			state.currentPreferencesState."$it.key".status = "disablePending"
+		} else {
+			state.currentPreferencesState."$it.key".status = "synced"
+		}
+	}
+	readConfigurationFromTheDevice()
+	// Preferences template end
+}
+
+def updated() {
+// Device-Watch simply pings if no device events received for 32min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
```

Comment:
This device shouldn't be offline pingable. It's intended only for life safety devices.